### DISCUSSION TITLE
Are we going to allow the user to add special constraints?

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#486] [Questions] [answered]
It scares me a bit to give users the freedom to mess up the model, but it will probably be necessary for some studies.

Example: 40% of hydrogen production has to be green.

**Comment by clizbe** [comment]
@datejada Is this covered by #526 ? Or are there crazier things people will add?

**Reply by datejada** [reply, marked as answer]
#526 covers the example you posted. I will only create a more complex user-defined constraint functionality if we start seeing that we need it because we can cover most of our use cases with what we have in the pipeline of developments.
